src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,186 @@
+        if (checksSubDirNames != null) {
+            final FilenameFilter filter = (rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            };
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(PACKAGE_NAME + "/" + dirName);
+                final String[] fileNames = dirFile.list(filter);
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list(filter);
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : checkHandler.list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
+            final String errorMessage = "URL in " + AVAILABLE_CHECKS_PATH + " is inconsistent: "
+                    + moduleName + " " + checkName1 + " " + checkName2;
+            assertWithMessage(errorMessage).that(checkName2).matches(checkName1);
+            if (checkName1.equals(COMMENTS_INDENTATION) || checkName1.equals(INDENTATION)) {
+                assertWithMessage(errorMessage).that(moduleName).matches(MISC);
+            }
+            else if (checkName1.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertWithMessage(errorMessage).that(moduleName).matches(ANNOTATION);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(moduleName);
+                if (moduleFileNames != null) {
+                    boolean match = false;
+                    final String checkNameWithSuffix = checkName1 + SUFFIX_CHECK_JAVA;
+                    for (String filename : moduleFileNames) {
+                        if (checkNameWithSuffix.equals(filename)) {
+                            match = true;
+                            break;
+                        }
+                    }
+                    assertWithMessage(errorMessage).that(match).isTrue();
+                }
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_ONE = "#";
+
+        public static final String SPLIT_TWO = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private final String nodeName;
+
+        private List<List<String>> list;
+
+        private List<String> stringList;
```

Comment:
please rename variables. It is confusing to see `list` and `stringList`, it is not clear what they are for

Reply:
done